internal/creds.go

```diff
@@ -102,16 +97,35 @@ func credentialsFromJSON(ctx context.Context, data []byte, ds *DialSettings) (*g
 	return cred, err
```

Comment:
The code flow is a bit inconsistent with the function comment. Function comment implies there are only two cases, OAuth and SelfSignedJWT, yet in code there is currently a 3rd case where neither OAuth or JWTFlow is true, and the cred is straight-up returned, which is the same behavior as when OAuth is true. Should we re-arrange the code flow to make isJWTFlow the first "if", and the rest can straight-up fall back to OAuth? (In other words, isSelfSignedJWTFlow can be updated with the logic in isOAuthFlow and isOAuthFlow function can be removed.)